.github/workflows/deploy_functions.yml

```diff
@@ -42,27 +49,54 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
+          ref: ${{ github.ref_name }}
           fetch-depth: 5
+
       - name: Cache node_modules
         uses: actions/cache@v2
         with:
           path: '**/node_modules'
           key: modules-${{ hashFiles('./yarn.lock') }}
 
+      - name: Generate and publish release
+        if: ${{ inputs.environment }} == 'development'
+        uses: release-drafter/release-drafter@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          publish: true
+
+      - name: Set Enviroment Variables
+        run: |
+          case ${{ inputs.environment }} in
```

Comment:
Just an idea, it might be kind of overengineer. But you could extract these "Set Environment Variables" to a third action and avoid code duplication across these two actions. What do you think?